crates/fs/private/key.rs

```diff
@@ -0,0 +1,108 @@
+use std::fmt::Debug;
+
+use aes_gcm::aead::{Aead, NewAead};
+use aes_gcm::{Aes256Gcm, Key as AesKey, Nonce};
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use crate::FsError;
+
+use super::Rng;
+
+//--------------------------------------------------------------------------------------------------
+// Contants
+//--------------------------------------------------------------------------------------------------
+
+pub(crate) const NONCE_SIZE: usize = 12;
+
+//--------------------------------------------------------------------------------------------------
+// Type Definitions
+//--------------------------------------------------------------------------------------------------
+
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Key(pub(super) [u8; 32]);
+
+//--------------------------------------------------------------------------------------------------
+// Implementations
+//--------------------------------------------------------------------------------------------------
+
+impl Key {
+    /// Creates a new key from [u8; 32].
+    pub fn new(bytes: [u8; 32]) -> Self {
+        Self(bytes)
+    }
+
+    /// Encrypts the given plaintext using the key.
+    pub fn encrypt(&self, nonce_bytes: &[u8; NONCE_SIZE], data: &[u8]) -> Result<Vec<u8>> {
+        let nonce = Nonce::from_slice(nonce_bytes);
+
+        let cipher_text = Aes256Gcm::new(AesKey::from_slice(&self.0))
+            .encrypt(nonce, data)
+            .map_err(|e| FsError::UnableToEncrypt(format!("{}", e)))?;
+
+        Ok([cipher_text, nonce_bytes.to_vec()].concat())
```

Comment:
Usually you *pre*pend the nonce. That would allow an easier implementation of a streaming decryption as well.
([This stackoverflow answer](https://crypto.stackexchange.com/a/70855) explains my point much better.)